pkg/server/server.go

```diff
@@ -532,3 +541,79 @@ func goContext(parent genericapiserver.PostStartHookContext) context.Context {
 func (s *Server) WaitForPhase1Finished() {
 	<-s.rootPhase1FinishedCh
 }
+
+func (s *Server) leaderElectAndStartControllers(ctx context.Context) {
+	logger := klog.FromContext(ctx)
+
+	leaderElectionConfig := rest.CopyConfig(s.GenericConfig.LoopbackClientConfig)
+	leaderElectionConfig = rest.AddUserAgent(leaderElectionConfig, "kcp-leader-election")
+	// TODO(embik): is there a better way to access / generate the system:admin workspace url?
```

Comment:
after rebase we have a `LocalAdminCluster` constant.

Today we have [pkg/genericcontrolplane/localadmin/cluster.go](https://github.com/kcp-dev/kubernetes/blob/c790f4246d25fa4d2787d5ac815c863cc56cdc56/pkg/genericcontrolplane/localadmin/cluster.go#L20). Witht that `localadmin.Cluster.Path().Path()` gives you `/clusters/system:admin`.

Reply:
Went with `localadmin.Cluster.Path().RequestPath()`, can replace it if the rebase lands before this PR. I've also moved setting up the `leaderElectionConfig` outside of this function to ensure I can properly join URLs and fail the post-start-hook if it doesn't work.